### ISSUE TITLE
X_Classifier.predict_proba returns array of shape (n_samples, 1)  instead of (n_samples, 2).

### ISSUE THREAD

**Issue by lashdk** [xplainable/xplainable#125]
**Description of Issue:**
predict_proba method implemented in scikit-learn Estimators and LGBM usually returns returns the predicted probability for each class for each sample. XClassifier deviates from this behaviour returning only probabilities estimates for class 1.

**How to reproduce:**
```
from xplainable.core.models import XClassifier
from sklearn.datasets import make_classification
import numpy as np
X, y = make_classification(n_samples=1000,n_features=4,random_state=42,n_classes=2)
X = pd.DataFrame(X,columns=['Feat_'+str(i) for i in range(4)])
y = pd.Series(y)
print(f'Shape of training data {X.shape}')
print(f'Shape of target {y.shape}')

x_model = XClassifier()
x_model.fit(X, y)
print(x_model.predict_proba(X).shape)
```

**Screenshot**
![image](https://github.com/xplainable/xplainable/assets/66780681/64f4b806-44cd-4309-960e-3b03f6d49ead)

**Comment by LordVenter** [comment]
XClassifier only predicts between a 'positive/class 1' and 'negative/ class 0' value, so it will only give 1 probability, the other will naturally be 1-prob(class 1)

**Comment by lashdk** [comment]
@LordVenter I was hoping we could make this consistent like in sklearn, LGBM, XGBoost as a lot of pre-defined hyper parameter optimisation pipelines would expect predict_proba to return shape as (n_sample, n_classes)

**Comment by Jtuppack** [comment]
Yeah @lashdk I agree, we should definitely implement this because it's steps towards multiclass classification (n_sample, n_classes)
